buildchain/buildchain/deps.py

```diff
@@ -0,0 +1,36 @@
+from doit.exceptions import TaskError
+from pathlib import Path
+from typing import Optional
+import shutil
+from buildchain import config
+
+
+def task_check_for():
+
+    def get_command_location(command_name: str, binary_path: Optional[str]
+                             ) -> Optional[TaskError]:
+        """Search the location of the command `command_name`.
+
+        If the user already specified a path for it, check for the existence
+        of the binary at the given location. Otherwise, look it up from PATH.
+
+        If the binary is found, the `CMD` dict is updated, otherwise it fails.
+        """
+        kwargs = {'path': Path(binary_path).parent} if binary_path else {}
+        cmd_path = shutil.which(command_name, **kwargs)
+        if cmd_path is None:
+            return TaskError(msg='command {} not found in {}'.format(
+                command_name, Path(binary_path).parent
+                if binary_path else '$PATH'
+            ))
+        config.CMD[command_name] = cmd_path
+        # return None
+    for cmd, binary_path in config.CMD.items():
+        yield {
+            'name': cmd,
+            'doc': 'Check the presence of the {} command.'.format(cmd),
+            'actions': [(get_command_location, [cmd, binary_path], {})],
+            'file_dep': [],
+            'task_dep': [],
+            'uptodate': [False],
+        }
```

Comment:
You need to add `__all__ = utils.export_only_tasks(__name__)` at the end of the file, to only export the tasks (it's also done for the others, see `iso.py` for instance).